### ISSUE TITLE
Filter referring to policies raises database error (Postgres 16)

### ISSUE THREAD

**Issue by carlo-salinari** [directus/directus#23838]
### Describe the Bug

I've defined a custom Read policy on collection directus_user as follows:

![image](https://github.com/user-attachments/assets/a871d197-dc04-49bb-960c-10496b4afa73)

#### Intent of the policy

Let all users who have policy Local Admin see in their User Directory:
- themselves
- all users with role Global Admin
- all users who have at least one policy in common

#### Expected behavior
User see himself, all users with role Global Admin and all users who have at least one policy in common with this user.

#### Observed behavior
No users are shown.
System Logs records a database Error:

![image](https://github.com/user-attachments/assets/02a179f6-345b-4587-bd6b-7512d79580b5)

Log dump
[db_error.log](https://github.com/user-attachments/files/17331983/db_error.log)

### To Reproduce

Basically, I get a db error every time I try to interact with the policies collection inside the filter.
Here's a view of Local Admin Permissions:

![image](https://github.com/user-attachments/assets/3a0c2c8a-bb96-4fb4-81f3-dbc425aa3b23)

### Directus Version

v11.1.1

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

Postgres 16.4

**Comment by hanneskuettner** [comment]
Can you provide us with the full error and paste it in a code block. The partial screenshot does not tell the full story sadly.

**Comment by carlo-salinari** [comment]
> Can you provide us with the full error and paste it in a code block. The partial screenshot does not tell the full story sadly.

I had attached the error in file db_error.log (see above).

```

{
  "level": 50,
  "time": 1728581395158,
  "msg": "select \"directus_users\".\"role\", CASE WHEN \"inner\".\"tzvta_email\" > 0 THEN \"directus_users\".\"email\" END as \"email\", CASE WHEN \"inner\".\"tzvta_id\" > 0 THEN \"directus_users\".\"id\" END as \"id\", CASE WHEN \"inner\".\"tzvta_policies\" > 0 THEN 1 END as \"policies\" from \"directus_users\" inner join (select \"directus_users\".\"id\", \"drxkp\".\"name\" as \"sort_arfvg\", COUNT((CASE WHEN (((\"directus_users\".\"id\" = $1 or \"drxkp\".\"name\" = $2 or \"sddrr\".\"id\" in ($3, $4))) or \"directus_users\".\"id\" = $5) THEN 1 END)) AS \"tzvta_email\", COUNT((CASE WHEN (((\"directus_users\".\"id\" = $6 or \"drxkp\".\"name\" = $7 or \"sddrr\".\"id\" in ($8, $9))) or \"directus_users\".\"id\" = $10) THEN 1 END)) AS \"tzvta_id\", COUNT((CASE WHEN (((\"directus_users\".\"id\" = $11 or \"drxkp\".\"name\" = $12 or \"sddrr\".\"id\" in ($13, $14)))) THEN 1 END)) AS \"tzvta_policies\" from \"directus_users\" left join \"directus_roles\" as \"drxkp\" on \"directus_users\".\"role\" = \"drxkp\".\"id\" left join \"directus_access\" as \"sznwd\" on \"directus_users\".\"id\" = \"sznwd\".\"user\" left join \"directus_policies\" as \"sddrr\" on \"sznwd\".\"policy\" = \"sddrr\".\"id\" where ((\"directus_users\".\"role\" = $15) and (((\"directus_users\".\"id\" = $16 or \"drxkp\".\"name\" = $17 or \"sddrr\".\"id\" in ($18, $19))) or \"directus_users\".\"id\" = $20)) group by \"directus_users\".\"id\" order by \"drxkp\".\"name\" asc limit $21) as \"inner\" on \"directus_users\".\"id\" = \"inner\".\"id\" order by \"inner\".\"sort_arfvg\" asc - column \"drxkp.name\" must appear in the GROUP BY clause or be used in an aggregate function"
}


```

**Comment by hanneskuettner** [comment]
Whoops sorry I totally missed that! Thank you!

**Comment by domdus** [comment]
@hanneskuettner I linked [#23862](https://github.com/directus/directus/issues/23862) as it might be related.
